resource.cpp

```diff
@@ -1178,13 +1190,28 @@ bool ResourceItem::setValue(const QVariant &val, ValueSource source)
             return true;
         }
     }
-    else
+    else if (m_rid->type == DataTypeReal)
     {
-        if (m_rid->type == DataTypeReal)
+        bool ok = false;
+        double d = val.toDouble(&ok);
+
+        if (ok)
         {
-            DBG_Printf(DBG_ERROR, "todo handle DataTypeReal in %s", __FUNCTION__);
-        }
+            m_lastSet = now;
+            m_doublePrev = m_double;
+            m_flags |= FlagNeedPushSet;
 
+            if (m_double != d)
```

Comment:
The exact comparison of floating point values can be a bit tricky at times, but might be ok in this case.

Another thing that we should be aware of are NaN, +-Infinity and negative zero values. Not sure if we need to have any limitations here.

Reply:
I don’t think we have any API writeable `DataTypeReal` items yet, so these non-real values would have to come from the Zigbee attribute through the JS `eval`.  Once we’ll do `messured_value` in CLIP sensors, we would need a check to only specify real values in the API, setting the item.